### ISSUE TITLE
adjectival present participles.

### ISSUE THREAD

**Issue by linas** [opencog/link-grammar#1233]
The following is parsed incorrectly:
```
He grew up speaking English
```

"grow up" is an intransitive verb, yet the parser recognizes "English" as the object of this sentence and "speaking" as an adjective describing English erroneously.

Reported by Stephen Frechette 14 June 2021 via email

**Comment by linas** [comment]
More examples:
```
After a long absence, he showed up speaking English
he arrived speaking English
```
The the following seem similar, but are quite different, and they do work correctly. Note the comma breaks these up into two distinct sentences.
```
he arrived, playing music
he departed, whistling to himself
he celebrated by lifting his glass
he cheered, lifting his glass
```

**Comment by stephenfrechette** [comment]
What the parser seems to be doing is mistaking adverbial gerunds with objects for adjectival present participles modifying an object of the sentence (even for intransitive verbs).
"Adverbial gerunds" might be a better title for this issue.

**Comment by linas** [comment]
Any suggestions for what the correct linkage should be?  I'm guessing
```
    +---->WV---->+------XXX-----+
    +->Wd--+--Ss-+---K--+       +---YYY--+
    |      |     |      |       |        |
LEFT-WALL he grew.v-d up.r speaking.g English
```
but its not clear what `XXX` and `YYY` should be.   I guess `YYY` should be 'O` because of `I am speaking English`.  It seems that a new link type for `XXX` might need to be invented. The `AG` link type?  Or maybe a new `MV` subtype, say `MVg` ?

The number of verbs that would take it seems slim. So far, I have this list:
* grew up
* woke up  -- `I woke up thinking of her`
* went out -- `we went out drinking beer`

Any others? Most other examples I can think of are really two distinct sentences, with a missing comma (e.g. `he departed, whistling to himself` -- with the comma, its clearly two distinct sentences. Without the comma, its ... ???)

**Comment by stephenfrechette** [comment]
It appears MVg is already a linkage.  It does not appear to be documented?
After some playing around, I found the following parses correctly (with an MVg linkage):
`The swan swam, causing waves to form.`

**Comment by stephenfrechette** [comment]
Is there a distinction between the Pg linkage and the MVg linkage in this case?  They seem to have the same function, perhaps they should be consolidated?

**Comment by stephenfrechette** [comment]
A few more examples that link differently in the current version:
`He sat there crying` (interprets "crying" as an object, I don't think this makes sense)
`He went running` (Parser uses a Pg linkage)
I now realize all of these examples could be used with any adjective, so perhaps these ARE adjectival participles, and NOT gerunds.
`He passed away happy`
`He died lonely`
`The swan swam fast`
`He sat there naked`
`He went crazy`

Some of these parse with variants of the MV linkage, some with variants of the P linkage, and some not at all.

So, this seems to not be an issue with gerunds or participles, but in constructions with adjectives acting adverbially?

**Comment by linas** [comment]
Hmm. OK, yes, looks like in most cases, `Pg` should be used.  

There are still some confusing cases. 
* `he swam, leaving a wake` -- the swimming is causing the wake, so `Pg`
* `he sat, crying` -- the sitting is not causing the crying; its a distinct activity. so no `Pg` link. ??
* `he cheered, lifting his glass` -- the cheering is not causing the lifting; a distinct activity.  To `Pg` or not `Pg` ??
* `he grew up speaking English` --- a comma is NOT allowed.  The growing is not causing the speaking, but its deeply tangled in.  So, sure, a `Pg` link here.

Welcome to my land of confusion.

**Comment by stephenfrechette** [comment]
I agree with your confusion here, this originally started due to what I thought were adverbial gerunds, but it turns out they were actually adverbial adjectival verbals!

Some of these seem similar to constructions with a "while" preposition omitted from them,
such as
`He sat, while crying`
`He cheered, while lifting his glass`
`He grew up while speaking English ??`
But then what is
`He went running`
in that case?  Is this causal?  None of this is very clear to me currently.

**Comment by linas** [comment]
I worked on this a bit, but have to put it down for the day.  Here is what I found so far.
* The `Pg` links is to be used when the present participle is mandatory. For example, "I enjoy running", vs. "*I enjoy".
* The `MVg` can be used when the present participle is used as a modifier. For example,  "it swam away, leaving a wake"
* The current rules in LG force a comma to be used; without the comma "it swam away leaving a wake" fails to parse. I tried to make the comma optional, but this leads to massive failures of the unit tests: lots of nonsense sentences suddenly become parsable.
* The above problem points at a design flaw. The `@MV` link appears all over the place. The comma was used as a crutch, to tighten the constraints. I don't like this but don't see any easy solution right now (other than to handle the comma with a phantom, as discussed in #224)

Thus the sentences "he grow up speaking English" and "we went out drinking beer" remain problematic. I don't want to use `Pg` because the participle is not mandatory. I can't use `MVg` because of the comma issue. Note that both sentences parse OK, if a comma is inserted: "he grew up, speaking English".
